src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -806,3 +809,18 @@ def __set_errors_json(self, error_count_by_operation, errors_by_operation):
         }
     # endregion
 
+    def __removed_older_complete_status_files(self, status_folder):
+        """ Retain the latest status complete file but remove older .complete.status files """
+        files_to_be_removed = []
+        list_of_files = glob.glob(status_folder + '/' + '*.complete.status')
+
+        for file in list_of_files:
+            if file != self.complete_status_file_path:
+                files_to_be_removed.append(os.path.basename(file))
+
+        if len(files_to_be_removed) < 1:
+            return
+
```

Comment:
We need to retain at least (configurable value) 10 older complete status files for troubleshooting purposes.

Retaining just the latest will not help in those situations.

Reply:
I've modified the logic to retain minimum 10 complete status file and use the constant for that

Reply:
done